internal/common/objects/MegaBike.go

```diff
@@ -81,10 +82,15 @@ func (mb *MegaBike) UpdateForce() {
 // Calculates the final orientation of the Megabike, between -1 and 1 (-180° to 180°), given the Biker's Turning forces
 func (mb *MegaBike) UpdateOrientation() {
 	totalTurning := 0.0
+	numOfSteeringAgents := 0.0
 	for _, agent := range mb.agents {
-		totalTurning += float64(agent.GetForces().Turning)
+		// if agents do not want to steer, they must set their turning force to math.NaN()
+		if math.IsNaN(totalTurning) {
+			numOfSteeringAgents += 1.0
+			totalTurning += float64(agent.GetForces().Turning)
+		}
 	}
-	averageTurning := totalTurning / float64(len(mb.agents))
```

Comment:
I think here there is a typo and it's meant to check whether the turning force is NaN to check whether an agent wants to submit it or not (so that the number of steering  agents is not actually the length of the map). @WillPowellUk can you confirm whether you meant math.IsNaN(agent.GetForces().Turning) instead of math.IsNaN(totalTurning)?